tsdb/meta.go

```diff
@@ -780,56 +781,43 @@ func (m *Measurement) TagSets(dimensions []string, condition influxql.Expr) ([]*
 		for _, dim := range dimensions {
 			tags[dim] = s.Tags.GetString(dim)
 		}
-
 		// Convert the TagSet to a string, so it can be added to a map allowing TagSets to be handled
 		// as a set.
-		tagsAsKey := string(MarshalTags(tags))
-		tagSet, ok := tagSets[tagsAsKey]
+		tagsAsKey := MarshalTags(tags)
+		tagSet, ok := tagSets[string(tagsAsKey)]
```

Comment:
@jsternberg yeah this approach is significantly than the previous approach. Around `15ns` per lookup versus `45ns` from some recent benchmarking I did.